Packages/com.unity.inputsystem/InputSystem/Editor/Settings/InputSettingsProvider.cs

```diff
@@ -185,11 +166,16 @@ private void InitializeWithCurrentSettings()
             var currentSettingsPath = AssetDatabase.GetAssetPath(m_Settings);
             if (string.IsNullOrEmpty(currentSettingsPath))
             {
-                // The current settings aren't coming from an asset. These won't be editable
-                // in the UI but we still have to show something.
-
-                m_CurrentSelectedInputSettingsAsset = ArrayHelpers.Append(ref m_AvailableInputSettingsAssets, "<No Asset>");
-                EditorBuildSettings.RemoveConfigObject(kEditorBuildSettingsConfigKey);
+                if (m_AvailableInputSettingsAssets.Length != 0)
+                {
+                    m_CurrentSelectedInputSettingsAsset = 0;
+                    m_Settings = AssetDatabase.LoadAssetAtPath<InputSettings>(m_AvailableInputSettingsAssets[0]);
+                    InputSystem.settings = m_Settings;
+                }
+                else
+                {
+                    CreateNewSettingsAsset("Assets/InputSystem.inputsettings.asset");
```

Comment:
Hmm, this one I'm not so sure about. To me it seems that just *looking* at something should not have the side effect of creating an asset. But *if* I read the changes correctly, then just clicking "Input (NEW)" in the project settings now creates an asset. As a user, that would bother me.

Reply:
In a similar vein, it seems pointless to create an asset if all it does is have the default settings. To me, it seemed nice to basically make the input settings asset thing only something you needed to bother with if you *actually* needed it.

Reply:
Ok, added a button as you suggested :)